src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -651,7 +651,33 @@ private BoundDecisionDag MakeBoundDecisionDag(SyntaxNode syntax, ImmutableArray<
 
             var rootDecisionDagNode = decisionDag.RootNode.Dag;
             RoslynDebug.Assert(rootDecisionDagNode != null);
-            return new BoundDecisionDag(rootDecisionDagNode.Syntax, rootDecisionDagNode);
+            var boundDecisionDag = new BoundDecisionDag(rootDecisionDagNode.Syntax, rootDecisionDagNode);
+#if DEBUG
+            int nextTempNumber = 0;
+            // Note that this leverages the `BoundDagEvaluation.GetHashCode()`
```

Comment:
nit: I think both GetHashCode and the Equals methods are needed tto make this work.
Consider "// Note we customize equality in BoundDagEvaluation"

Reply:
Fixed